cpu/esp32/vendor/esp-idf/wpa_supplicant/src/crypto/Makefile

```diff
@@ -11,6 +11,7 @@ CFLAGS += -D__ets__ -DESPRESSIF_USE
 
 include $(RIOTCPU)/$(CPU)/Makefile.include
 
+# Why are these includes here (this is a module)
```

Comment:
@gschorcht Same thing here. Why was it necessary to manually edit "INCLUDES"? I want to know if I can take out the `INCLUDES += -I$(RIOTBASE)/sys/posix/include`

Reply:
@jcarrano Unfortunatly, this is much more tricky. If you take a look at line 8, you will find
```
INCLUDES  = -I$(RIOTCPU)/$(CPU)/vendor/esp-idf/wpa_supplicant/port/include
```
verwrites all includes defined by ``` $(RIOTBASE)/Makefile.base```. This was necessary to change the order of include pathes. Otherwise, header files from ```$(RIOTBASE)/sys/include/crypto``` would be used instead of the header files from ```$(ESP32_SDK_DIR)/components/wpa_supplicant/include/crypto``` which are in conflict. Maybe there is a better approach, but I couldn't figure out one in the very complex make structure of RIOT when I was porting RIOT to ESP32.

I'm not really sure, I tried to comment out both include lines from ```$(RIOTBASE)/sys/```. It seems to be compilabe without them. Maybe, there not needed anymore and can be removed.

Reply:
Finally, I figured out what the reason for these ```INCLUDES``` was. Let's suppose you try to compile an application for ESP32 with activated WiFi netdev, e.g. the ```esp_now``` module, together with the ```spiffs``` package with command:
```
USEMODULE="spiffs esp_now" make BOARD=esp32-wroom-32 -C tests/shell_xx_esp32
```
- Package ```spiffs``` depends on modules ```mtd``` and ```vfs```.
- If module ```mtd``` is used, the board definition includes ```mtd.h``` to declare the default MTD device ```mtd0```. This is done in the same way in ```native``` and ```mulle``` board definitions.
- ```mtd.h``` includes ```vfs.h``` if module ```vfs``` is enabled.
- ```vfs.h``` includes ```sys/statvfs.h``` which is found in ```sys/posix/include/sys/statvfs.h```.

Because of conflicts between the header files ```$(ESP32_SDK_DIR)/components/wpa_supplicant/include/crypto``` used by the ESP32 WiFi modules and ```$(RIOTBASE)/sys/include/crypto```, we have to ensure that the include order is the following:
```
-I$(ESP32_SDK_DIR)/components/wpa_supplicant/include/crypto
-I$(RIOTBASE)/sys/include/
```
Unfortunately, I could not find a way to get the include paths of a module in front of RIOT's standard include paths in the ```INCLUDES``` variable. Therefore, I had to set ```INCLUDES``` for this module from scratch including ```$(RIOTBASE)/sys/posix/include``` :worried:

Maybe you, @cladmi or @kaspar030 have an idea how to deal with such a case.

Reply:
@gschorcht I have seen this inconsistency a while a go but did not have any applied reason to change it except "it looks weird", now there is one :)
For me it feels like the https://github.com/RIOT-OS/RIOT/blob/master/Makefile.include#L239 line should be after processing board and cpu Makefile.include, even after the `toolchain` and `newlib` headers (that basically trick this and overwrites the deferred value of `INCLUDES` https://github.com/RIOT-OS/RIOT/blob/98f42eeea8e04b0e2109aa77e04d51f566f60bee/makefiles/libc/newlib.mk#L84-L89) even put them first as BSP should go first.

Reply:
I just noticed that your issue is for one module makefile and that the header is required for this module only.
For me it is the same case as we have for the `NATIVEINCLUDES` where implementing the board specific modules require specific includes and do not even need to see RIOT headers.
You basically want to define a `ESP32_VENDORMODULE_INCLUDES`.